### PR TITLE
Move classes out of IoT Hub device client's "common" folder

### PR DESCRIPTION
**Pull request by timtay-microsoft**
These classes are moved to the most appropriate spot based on what set of folders contained classes that referenced them. Some classes were deleted because they were unused.

None of these classes were used outside of the IoT Hub device client package, so I'm not sure why this "common" folder existed in the first place

### CODE REVIEW

File: iothub/device/src/Transport/Amqp/AccessRights.cs [hunk 1]
```diff
@@ -6,7 +6,7 @@
 using Newtonsoft.Json;
 using Newtonsoft.Json.Converters;
 
-namespace Microsoft.Azure.Devices.Client
+namespace Microsoft.Azure.Devices.Client.Transport.Amqp
```
Comment (timtay-microsoft) [inline iothub/device/src/Transport/Amqp/AccessRights.cs:9]:
```suggestion
namespace Microsoft.Azure.Devices.Client
```

File: iothub/device/src/Transport/Http/CustomHeaderConstants.cs [hunk 2]
```diff
@@ -1,7 +1,7 @@
 // Copyright (c) Microsoft. All rights reserved.
 // Licensed under the MIT license. See LICENSE file in the project root for full license information.
 
-namespace Microsoft.Azure.Devices.Client
+namespace Microsoft.Azure.Devices.Client.Transport
```
Comment (drwill-ms) [inline iothub/device/src/Transport/Http/CustomHeaderConstants.cs:4]: I'd like to not have namespaces match the folders. Unless the files in there already do, can we leave this the same as the root namespace?

**Review by drwill-ms — APPROVED** [review]

File: iothub/device/src/Transport/Http/CustomHeaderConstants.cs (hunk 2 above)
Comment (timtay-microsoft) [reply iothub/device/src/Transport/Http/CustomHeaderConstants.cs:4]: Sure I can revert this change

**Pull request merged**